simexpal/build.py

```diff
@@ -264,12 +264,29 @@ def git_ref_changed():
 						build.clone_dir,
 						generic_tag])
 			else:
-				# Recreate the source directory
-				util.try_rmtree(build.source_dir)
-				util.try_mkdir(build.source_dir)
-
-				# Clone the git repository into the build.source_dir
-				subprocess.check_call(['git', 'clone', build.info.git_repo, build.source_dir])
+				branch = build.revision.version_for_build(build.name)
+				if os.path.isdir(build.source_dir) and not reclone:
+					# Repository already cloned, check for local changes
+					status = subprocess.check_output(['git', 'status', '--porcelain'], cwd=build.source_dir)
+					if len(status) != 0:
+						print("There are local changes that have not yet been commited and pushed. Please clean them up before proceeding.")
+						return
+
+					# or non-pushed commits
+					unpushed_commits = subprocess.check_output(['git', 'log', '--branches', '--not', '--remotes'], cwd=build.source_dir)
+					if len(unpushed_commits) != 0:
```

Comment:
I would say it is the best to not care about the unpushed commits. So this `if` clause can be removed.

Reply:
I've removed the check for unpushed commits.